pennylane/templates/tensornetworks/ttn.py

```diff
@@ -0,0 +1,210 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the TTN template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+import warnings
+import pennylane as qml
+import pennylane.numpy as np
+from pennylane.operation import Operation, AnyWires
+
+
+def compute_indices(wires, n_block_wires):
+    """Generate a list containing the wires for each block.
+
+    Args:
+        wires (Iterable): wires that the template acts on
+        n_block_wires (int): number of wires per block
+    Returns:
+        layers (array): array of wire indices or wire labels for each block
+    """
+
+    n_wires = len(wires)
+
+    if n_block_wires % 2 != 0:
+        raise ValueError(f"n_block_wires must be an even integer; got {n_block_wires}")
+
+    if n_block_wires < 2:
+        raise ValueError(
+            f"number of wires in each block must be larger than or equal to 2; got n_block_wires = {n_block_wires}"
+        )
+
+    if n_block_wires > n_wires:
+        raise ValueError(
+            f"n_block_wires must be smaller than or equal to the number of wires; "
+            f"got n_block_wires = {n_block_wires} and number of wires = {n_wires}"
+        )
+
+    if not np.log2(n_wires / n_block_wires).is_integer():
+        warnings.warn(
+            f"The number of wires should be n_block_wires times 2^n; got n_wires/n_block_wires = {n_wires/n_block_wires}"
+        )
+
+    n_wires = 2 ** (int(np.log2(len(wires) / n_block_wires))) * n_block_wires
+    n_layers = int(np.log2(n_wires // n_block_wires)) + 1
+
+    layers = np.array(
+        [
+            [
+                wires[i]
+                for i in range(
+                    x + 2 ** (j - 1) * n_block_wires // 2 - n_block_wires // 2,
+                    x + n_block_wires // 2 + 2 ** (j - 1) * n_block_wires // 2 - n_block_wires // 2,
+                )
+            ]
+            + [
+                wires[i]
+                for i in range(
+                    x
+                    + 2 ** (j - 1) * n_block_wires // 2
+                    + 2 ** (j - 1) * n_block_wires // 2
+                    - n_block_wires // 2,
+                    x
+                    + 2 ** (j - 1) * n_block_wires // 2
+                    + n_block_wires // 2
+                    + 2 ** (j - 1) * n_block_wires // 2
+                    - n_block_wires // 2,
+                )
+            ]
+            for j in range(1, n_layers + 1)
+            for x in range(0, n_wires - n_block_wires // 2, 2 ** (j - 1) * n_block_wires)
+        ]
+    )
```

Comment:
If this just contains wire indices, why not use a list or tuple?

Reply:
Thanks! I'll make this change.